go/adbc/driver/snowflake/statement.go

```diff
@@ -499,41 +523,56 @@ func (st *statement) ExecuteQuery(ctx context.Context) (array.RecordReader, int6
 		st.streamBind = nil
 
 		rdr := concatReader{}
-		err := rdr.Init(&bind)
+		err = rdr.Init(&bind)
 		if err != nil {
 			return nil, -1, err
 		}
 		return &rdr, -1, nil
 	}
 
-	loader, err := st.cnxn.cn.QueryArrowStream(ctx, st.query)
+	var loader gosnowflake.ArrowStreamLoader
+	loader, err = st.cnxn.cn.QueryArrowStream(ctx, st.query)
 	if err != nil {
-		return nil, -1, errToAdbcErr(adbc.StatusInternal, err)
+		err = errToAdbcErr(adbc.StatusInternal, err)
+		return nil, -1, err
 	}
 
-	rdr, err := newRecordReader(ctx, st.alloc, loader, st.queueSize, st.prefetchConcurrency, st.useHighPrecision)
-	nrec := loader.TotalRows()
-	return rdr, nrec, err
+	reader, err = newRecordReader(ctx, st.alloc, loader, st.queueSize, st.prefetchConcurrency, st.useHighPrecision)
+	nrows = loader.TotalRows()
+	return reader, nrows, err
 }
 
 // ExecuteUpdate executes a statement that does not generate a result
 // set. It returns the number of rows affected if known, otherwise -1.
-func (st *statement) ExecuteUpdate(ctx context.Context) (int64, error) {
+func (st *statement) ExecuteUpdate(ctx context.Context) (nrows int64, err error) {
 	ctx = st.setQueryContext(ctx)
+	err = nil
+	nrows = -1
+
+	_, span := st.StartSpan(ctx, "ExecuteUpdate")
```

Comment:
same comment as above. you can't ignore the `ctx` return from `st.StartSpan`, you should swap the order here. It should look like:

```go
var span trace.Span
ctx, span = st.StartSpan(ctx, "ExecuteUpdate")
ctx, nrows = st.setQueryContext(ctx), -1
defer func() { .....
```